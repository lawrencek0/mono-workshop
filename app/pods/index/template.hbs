<div class="row">
    <div class="col-md-1"></div>
    <div class="col-md-10">
        <div class="center">
            <h1 class="h1 text-uppercase text-center">Welcome to Nepal Portal!</h1>
            <p class="text-justify">
                Nepal Portal was developed with the main intention of helping the people by making it easier to find the
                petrol pumps where the fuel is being distributed in this hard time.
            </p>
            <!-- FLEXSLIDER AREA !-->
        </div>
    </div>
    <div class="col-md-1"></div>
</div>
