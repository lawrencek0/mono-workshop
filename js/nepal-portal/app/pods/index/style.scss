#mega-map {
  visibility: visible;
  display: block;
  width: 100%;
  height: 600px;
}
