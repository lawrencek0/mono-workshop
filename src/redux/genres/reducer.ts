import { combineReducers } from 'redux';
import { getType } from 'typesafe-actions';
import { genresActions } from './';

export interface Genre {
  readonly id: number;
  readonly name: string;
}

export type GenresState = {
  readonly isFetching: boolean;
  readonly errMessage: string | null;
  readonly genres: Genre[];
};

export type RootState = {
  readonly movies: GenresState;
};

export const genresReducer = combineReducers<GenresState>({
  isFetching: (state = false, action) => {
    switch (action.type) {
      case getType(genresActions.fetchGenres):
        return true;
      case getType(genresActions.fetchGenresFulfilled):
        return false;
      default:
        return state;
    }
  },
  errMessage: (state = '', action) => {
    switch (action.type) {
      default:
        return state;
    }
  },
  genres: (state = [], action) => {
    switch (action.type) {
      case getType(genresActions.fetchGenresFulfilled): {
        return action.payload;
      }
      default:
        return state;
    }
  }
});

import { $call } from 'utility-types';
const returnsOfActions = Object.values(genresActions).map($call);
export type GenresAction = typeof returnsOfActions[number];
