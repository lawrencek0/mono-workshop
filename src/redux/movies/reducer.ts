import { combineReducers } from 'redux';
import { getType } from 'typesafe-actions';
import { moviesActions } from './actions';

// @FIXME: underscore_case cause of the api any workarounds? reselct?
export interface Movie {
  id: number;
  title: string;
  genre_ids: number[];
  vote_count: number;
  vote_average: number;
  release_date: string;
  poster_path: string;
  className?: string;
}

export type MoviesState = {
  readonly isFetching: boolean;
  readonly errMessage: string | null;
  readonly movies: Movie[];
};

export type RootState = {
  movies: MoviesState;
};

export const moviesReducer = combineReducers<MoviesState>({
  isFetching: (state = false, action) => {
    switch (action.type) {
      case getType(moviesActions.fetchMovies):
        return true;
      case getType(moviesActions.fetchMoviesFullFilled):
        return false;
      default:
        return state;
    }
  },
  errMessage: (state = '', action) => {
    switch (action.type) {
      default:
        return state;
    }
  },
  movies: (state = [], action) => {
    switch (action.type) {
      case getType(moviesActions.fetchMovies):
        return state;

      case getType(moviesActions.fetchMoviesFullFilled): {
        return [...state, ...action.payload.results];
      }
      default:
        return state;
    }
  }
});

import { $call } from 'utility-types';
const returnsOfActions = Object.values(moviesActions).map($call);
export type MoviesAction = typeof returnsOfActions[number];
