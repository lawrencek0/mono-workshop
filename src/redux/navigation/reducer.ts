import { combineReducers } from 'redux';
import { getType } from 'typesafe-actions';
import { navigationActions } from './actions';
import * as config from './config';

export type SliderLabel = 'year' | 'rating' | 'runtime';

export interface SliderData {
  readonly label: SliderLabel;
  readonly min: number;
  readonly max: number;
  readonly step: number;
  readonly value: {
    readonly min: number;
    readonly max: number;
  };
}

export type NavigationState = {
  readonly selectedGenre: string;
  readonly year: SliderData;
  readonly rating: SliderData;
  readonly runtime: SliderData;
};

export type RootState = {
  readonly navigation: NavigationState;
};

export const navigationReducer = combineReducers<NavigationState>({
  selectedGenre: (state = config.DEFAULT_GENRE, action) => {
    switch (action.type) {
      case getType(navigationActions.updateSelectedGenre):
        return action.payload;
      default:
        return state;
    }
  },
  year: (state = config.DEFAULT_YEAR_CONFIG, action) => {
    switch (action.type) {
      case getType(navigationActions.updateYearSlider):
        return { ...state, value: action.payload };
      default:
        return state;
    }
  },
  rating: (state = config.DEFAULT_RATING_CONFIG, action) => {
    switch (action.type) {
      case getType(navigationActions.updateRatingSlider):
        return { ...state, value: action.payload };
      default:
        return state;
    }
  },
  runtime: (state = config.DEFAULT_RUNTIME_CONFIG, action) => {
    switch (action.type) {
      case getType(navigationActions.updateRuntimeSlider):
        return { ...state, value: action.payload };
      default:
        return state;
    }
  }
});
