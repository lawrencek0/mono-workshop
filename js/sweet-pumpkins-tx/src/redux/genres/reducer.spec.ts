import { genresReducer, Genre } from './reducer';
import { genresActions } from './index';

const genres: Genre[] = [
  {
    id: 28,
    name: 'Action'
  }
];
const initalState = genresReducer(undefined, {});

describe('genres reducer', () => {
  describe('initial state', () => {
    expect(initalState).toMatchSnapshot();
  });

  describe('adding genres', () => {
    it('should add a new genre', () => {
      const action = genresActions.fetchGenresFulfilled(genres);
      const state = genresReducer(initalState, action);
      // expect(state.genres).toHaveLength(1);
      // expect(state.genres[0].id).toEqual(action.payload[0].id);
    });
  });
});
