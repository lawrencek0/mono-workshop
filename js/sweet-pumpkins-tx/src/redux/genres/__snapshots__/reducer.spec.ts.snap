// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[` 1`] = `
Object {
  "errMessage": "",
  "genres": Array [],
  "isFetching": false,
}
`;
